### PR TITLE
💥 remove some polyfills

### PR DESCRIPTION
**Pull request by thomas-lebeau**
> [!Note]
> This PR is targeting the `v6` branch

## Motivation

Reduce bundle size by supporting only browser that support ES2018. We can remove some polyfills as they are supported by natively on the new browser support matrix

## Changes

Remove the following polyfills:
- Array.includes()
- String.includes()
- Array.find()
- Array.from()
- String.starstWith()
- String.endsWith()
- Array.forEach()
- CSS.escape()
- Element.matches()
- Node.parentElement()
- Element.classList
- WeakSet
- Element.closest()
- Element.labels
- Map.keys()

## Testing

- [x] Local
- [ ] Staging
- [x] Unit
- [x] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

**Comment by thomas-lebeau** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/DataDog/browser-sdk/2857?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2857** <a href="https://app.graphite.dev/github/pr/DataDog/browser-sdk/2857?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2856** <a href="https://app.graphite.dev/github/pr/DataDog/browser-sdk/2856?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `v6`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @thomas-lebeau and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/rum-core/src/domain/startCustomerDataTelemetry.ts [hunk 1]
```diff
@@ -71,7 +64,7 @@ export function startCustomerDataTelemetry(
 
     updateMeasure(
       currentBatchMeasures.featureFlagBytes,
-      includes([RumEventType.VIEW, RumEventType.ERROR], event.type)
+      [RumEventType.VIEW, RumEventType.ERROR].includes(event.type as any)
```
Comment (nulrich) [inline packages/rum-core/src/domain/startCustomerDataTelemetry.ts:67]:
```suggestion
      [RumEventType.VIEW, RumEventType.ERROR].includes(event.type as RumEventType)
```

File: packages/core/src/browser/pageExitObservable.ts [hunk 2]
```diff
@@ -51,6 +51,6 @@ export function createPageExitObservable(configuration: Configuration): Observab
   })
 }
 
-export function isPageExitReason(reason: string | undefined): reason is PageExitReason {
-  return includes(objectValues(PageExitReason), reason)
+export function isPageExitReason(reason: string): reason is PageExitReason {
+  return objectValues(PageExitReason).includes(reason as any)
```
Comment (nulrich) [inline packages/core/src/browser/pageExitObservable.ts:55]:
```suggestion
  return Object.values(PageExitReason).includes(reason as PageExitReason);
```

Comment (thomas-lebeau) [reply packages/core/src/browser/pageExitObservable.ts]: I kept `objectValues` for now, I plan to tackle them in another PR

File: packages/core/src/tools/utils/polyfills.ts [hunk 3]
```diff
@@ -50,33 +11,22 @@ export function findLast<T, S extends T>(
...
-export function endsWith(candidate: string, search: string) {
-  return candidate.slice(-search.length) === search
-}
-
 interface Assignable {
   [key: string]: any
 }
 
+// TODO: Object.assign generates larger bundle size than this polyfill.
+// try with spread operator after updating target to es2018
```
Comment (BenoitZugmeyer) [inline packages/core/src/tools/utils/polyfills.ts:29]: **💭 thought:** ‏Yes, we should use the spread operator and remove [those ESLint rules](https://github.com/DataDog/browser-sdk/blob/3abdb5bd13651276dac3947b822d4fbd24022c33/.eslintrc.js#L237-L244) and enable [prefer-object-spread](https://eslint.org/docs/latest/rules/prefer-object-spread#rule-details) once fully migrated to ES2018. Exciting times 😄

**Review by BenoitZugmeyer — APPROVED** [review]

File: packages/core/src/tools/utils/polyfills.ts (hunk 3 above)
Comment (thomas-lebeau) [reply packages/core/src/tools/utils/polyfills.ts:29]: spoiler: https://github.com/DataDog/browser-sdk/pull/2908

File: packages/rum-core/src/browser/polyfills.ts [hunk 4]
```diff
@@ -19,41 +20,10 @@ export function cssEscape(str: string) {
...
-  }
-
-  return null
-}
-
 /**
  * Return the classList of an element or an array of classes if classList is not supported
  *
- * In cases where classList is not supported, such as in IE11 for SVG and MathML elements,
+ * In cases where classList is not supported, such as in Opera Mini for SVG and MathML elements,
```
[the remark this answers is not in the document]
Comment (thomas-lebeau) [reply packages/rum-core/src/browser/polyfills.ts]: yes, It was an intermediary step while I was still trying to figure out what will be the browser matrix

**Review by nulrich — APPROVED** [review]

**Pull request merged**